### PR TITLE
[system][process] Add metric_type metadata for object fields, set stack restriction to 8.9.0 version

### PR DESCRIPTION
**Pull request by tetianakravchenko**
## What does this PR do?

It is a follow up of this PR comment - https://github.com/elastic/integrations/pull/6493#pullrequestreview-1467200573
to add all missing metric_type.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: packages/system/data_stream/process/fields/fields.yml [hunk 1]
```diff
@@ -688,16 +688,15 @@
...
                 Path to the cgroup relative to the cgroup subsystems mountpoint.
 
-            - name: stats.*
-              type: object
-              description: per-device IO usage stats
-            - name: stats.*.*
-              type: object
             - name: stats.*.*.bytes
               type: object
+              object_type: long
```
Comment (tetianakravchenko) [inline packages/system/data_stream/process/fields/fields.yml:693]: without adding `object_type` it fails during the `elastic-package build` with:
```
   1. file "/Users/tetianakravchenko/go/src/github.com/elastic/integrations/build/packages/system-1.39.0.zip/data_stream/process/fields/fields.yml" is invalid: field 0.fields.6.fields.5.fields.26.type: 0.fields.6.fields.5.fields.26.type must be one of the following: "histogram", "aggregate_metric_double", "long", "integer", "short", "byte", "double", "float", "half_float", "scaled_float", "unsigned_long"
```
those fields are detected as long, so I set `object_type: long` as well
Screenshot 2023-09-04 at 16 25 26

File: packages/system/changelog.yml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.39.0"
+  changes:
+    - description: Add metric_type metadata for object fields, set stack restriction to 8.9.0 version
+      type: bugfix
```
Comment (efd6) [inline packages/system/changelog.yml:5]: enhancement?

File: packages/system/data_stream/process/fields/fields.yml (hunk 1 above)
Comment (lalit-satapathy) [reply packages/system/data_stream/process/fields/fields.yml:693]: @tetianakravchenko, This is based on the [final syntax of dynamic mapping](https://github.com/elastic/kibana/issues/155004#issuecomment-1669221219?) we had agreed? Is there still any issues in 8.9?

Comment (tetianakravchenko) [reply packages/system/data_stream/process/fields/fields.yml:693]: yes, it is this syntax, there is no issues with 8.9

**Review by lalit-satapathy — COMMENTED** [review]
LGTM

File: packages/system/data_stream/process/fields/fields.yml [hunk 3]
```diff
@@ -541,11 +542,10 @@
               description: >
                 The number of times that the memory limit (kmem_tcp.limit.bytes) was reached.
 
-            - name: stats.*
```
Comment (lalit-satapathy) [inline packages/system/data_stream/process/fields/fields.yml:544]: @tetianakravchenko, Any reason we are removing the mapping in multiple places for  `stats.*` and we are only keeping the ones with `stats.*.bytes`

How will be the following field be mapped: `system.process.cgroup.memory.stats.page_faults`

Can we check the mapping of all the `stats` group under `system.process` ?

Comment (tetianakravchenko) [reply packages/system/data_stream/process/fields/fields.yml:544]: @lalit-satapathy I am not sure why we initially had `stats.*` format.

all metrics that includes `*` in the name are included now in `dynamic_templates`:
Screenshot 2023-09-25 at 16 44 26

> How will be the following field be mapped: system.process.cgroup.memory.stats.page_faults

`system.process.cgroup.memory.stats.page_faults` is defined here - https://github.com/elastic/integrations/blob/87e0e38053238c89e74a488a852a9d086dab97a4/packages/system/data_stream/process/fields/fields.yml#L602-L606 
Screenshot 2023-09-25 at 16 52 23

> Can we check the mapping of all the stats group under system.process ?

<details>

<summary> Stats object mapping under `system.process.cgroup.memory` </summary>

```
 "stats": {
                          "properties": {
                            "active_anon": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "active_file": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "anon": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "anon_thp": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "cache": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "file": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "file_dirty": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "file_mapped": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "file_thp": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "file_writeback": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "hierarchical_memory_limit": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "hierarchical_memsw_limit": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "htp_collapse_alloc": {
                              "type": "long"
                            },
                            "inactive_anon": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "inactive_file": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "kernel_stack": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "major_page_faults": {
                              "type": "long",
                              "time_series_metric": "counter"
                            },
                            "mapped_file": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "page_activate": {
                              "type": "long"
                            },
                            "page_deactivate": {
                              "type": "long"
                            },
                            "page_faults": {
                              "type": "long",
                              "time_series_metric": "counter"
                            },
                            "page_lazy_free": {
                              "type": "long"
                            },
                            "page_lazy_freed": {
                              "type": "long"
                            },
                            "page_refill": {
                              "type": "long"
                            },
                            "page_scan": {
                              "type": "long"
                            },
                            "page_steal": {
                              "type": "long"
                            },
                            "page_tables": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "pages_in": {
                              "type": "long",
                              "time_series_metric": "counter"
                            },
                            "pages_out": {
                              "type": "long",
                              "time_series_metric": "counter"
                            },
                            "per_cpu": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "rss": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "rss_huge": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "shmem": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "shmem_thp": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "slab": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "slab_reclaimable": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "slab_unreclaimable": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "sock": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "swap": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "swap_cached": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "thp_fault_alloc": {
                              "type": "long"
                            },
                            "unevictable": {
                              "properties": {
                                "bytes": {
                                  "type": "long",
                                  "time_series_metric": "gauge"
                                }
                              }
                            },
                            "workingset_activate_anon": {
                              "type": "long"
                            },
                            "workingset_activate_file": {
                              "type": "long"
                            },
                            "workingset_node_reclaim": {
                              "type": "long"
                            },
                            "workingset_refault_anon": {
                              "type": "long"
                            },
                            "workingset_refault_file": {
                              "type": "long"
                            },
                            "workingset_restore_anon": {
                              "type": "long"
                            },
                            "workingset_restore_file": {
                              "type": "long"
                            }
                          }
                        }
```

</details>

One thing I noticed that some fields are not defined in the `fields` folder and for this reason they are missing `time_series_metric`. I think it is fields that were added in this PR - https://github.com/elastic/beats/pull/27242.

I think there should be another PR for adding missing fields, to not mix it up in this PR

**Comment by lalit-satapathy** [comment]
>all metrics that includes * in the name are included now in dynamic_templates:

It will be good, if we confirm that no fields are left un-mapped because we removed  `stats.*`, `stats.*.*` mappings. Otherwise LGTM.

File: packages/system/data_stream/process/fields/fields.yml (hunk 3 above)
Comment (tetianakravchenko) [reply packages/system/data_stream/process/fields/fields.yml:544]: Hey @fearful-symmetry I see that this field was introduced in this PR - https://github.com/elastic/integrations/pull/1539 by you, could you please check if I am missing here something? I don't really understand why this was initially introduced, in addition to the `stats.*.bytes` - https://github.com/elastic/integrations/pull/1539/files#diff-c594e593105a59a8530cec1e0670442dbe7b2cc59074900a65b96e6b4f95352fR481-R483

File: packages/system/data_stream/process/fields/fields.yml [hunk 4]
```diff
@@ -688,16 +688,15 @@
...
-              type: object
-              description: per-device IO usage stats
-            - name: stats.*.*
-              type: object
             - name: stats.*.*.bytes
               type: object
+              object_type: long
+              metric_type: gauge
               description: per-device IO usage stats
```
Comment (ritalwar) [inline packages/system/data_stream/process/fields/fields.yml:695]: The description is the same as for `stats.*.*.ios`. What distinguishes it here?

**Review by ritalwar — APPROVED** [review]
LGTM!

**Comment by tetianakravchenko** [comment]
@lalit-satapathy 

> It will be good, if we confirm that no fields are left un-mapped because we removed stats.*, stats.*.* mappings. Otherwise LGTM.

can you please explain how do you want me to confirm it?
I think it is not even possible to get unmapped fields since by default is used [dynamic mapping](https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-field-mapping.html)

I've checked the latest system package version running in cloud (so without my changes):
Screenshot 2023-09-27 at 12 11 06


so fields `stats.*` are mapped either as defined in fields file - example `page_faults` https://github.com/elastic/integrations/blob/main/packages/system/data_stream/process/fields/fields.yml#L602-L625 or defined by the dynamic mapping - note that some fields like `page_activate` are not defined in the fields file, but are exposed by the beats - https://github.com/elastic/beats/blob/main/metricbeat/module/system/process/_meta/data.json#L170

On my local setup with the version that include changes in this PR:
Screenshot 2023-09-27 at 12 19 13

you can see that both mentioned fields are present as well - `page_activate` (not defined in list of fields) and `page_faults` (defined in list of fields)

**Review by lalit-satapathy — APPROVED** [review]
LGTM

**Pull request merged**